custom-targets/git-ops/quickstart/QUICKSTART.md

```diff
@@ -0,0 +1,171 @@
+# Cloud Deploy Git Deployer Quickstart
+
+This page shows you how to use Cloud Deploy to deploy to a Git Repository on `github.com` using the sample Git deployer.
+
+In this quickstart you will:
+
+1. Create a Github repository and a Personal Access Token to give a service account access to the repository.
+2. Define a Cloud Deploy delivery pipeline, custom target type for Git, and two targets (dev and prod).
+3. Create a Cloud Deploy release and rollout to write the rendered manifest to the Git repository under a `dev` path in the `deploy` branch and create a pull request to the `main` branch.
+4. Promote the release to deploy the rendered manifest to the Git repository under a `prod` path in the `deploy` branch and create a pull request to the `main` branch.
+
+## 1. Clone Repository
+
+Clone this repository and navigate to the quickstart directory (`cloud-deploy-samples/custom-targets/git-ops/quickstart`) since the commands provided expect to be executed from that directory.
+
+## 2. Environment Variables
+
+To simplify the commands in this quickstart, set the following environment variables with your values:
+
+```shell
+export PROJECT_ID="YOUR_PROJECT_ID"
+export REGION="YOUR_REGION"
+```
+
+## 3. Prerequisites
+
+### APIs
+Enable the Cloud Deploy API and Secret Manager API.
+
+```shell
+gcloud services enable clouddeploy.googleapis.com secretmanager.googleapis.com container.googleapis.com --project $PROJECT_ID
+```
+
+### Permissions
+Make sure the default compute service account, `{project_number}-compute@developer.gserviceaccoumt.com`, used by Cloud Deploy has sufficient permissions:
+
+1. `clouddeploy.jobRunner` role:
+
+    ```shell
+    gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member=serviceAccount:$(gcloud projects describe $PROJECT_ID \
+    --format="value(projectNumber)")-compute@developer.gserviceaccount.com \
+    --role="roles/clouddeploy.jobRunner"
+    ```
+
+2. `secretmanager.secretAccessor` role:
+
+    ```shell
+    gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member=serviceAccount:$(gcloud projects describe $PROJECT_ID \
+    --format="value(projectNumber)")-compute@developer.gserviceaccount.com \
+    --role="roles/secretmanager.secretAccessor"
+    ```
+
+## 4. Build and Register a Custom Target Type for Git
+From within the `quickstart` directory, run the following command to build the Git deployer image and register a Cloud Deploy custom target type that references the image:
+
+```shell
+../build_and_register.sh -p $PROJECT_ID -r $REGION
+```
+
+For information about the `build_and_register.sh` script, see the [README](../README.md#build)
+
+## 5. Create Github Repository and Personal Access Token (PAT)
+
+Create a [new Github repository](https://github.com/new) where the manifests will be written to when deploying. Set the repository and owner as an environment variables:
```

Comment:
My new GitHub™ repo didn't have a `main` branch, which broke things. Maybe tell people to check the "Add a README file" box, which will create it.